Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.hxx

```diff
@@ -0,0 +1,177 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+/*=========================================================================
+ *
+ *  Portions of this file are subject to the VTK Toolkit Version 3 copyright.
+ *
+ *  Copyright (c) Ken Martin, Will Schroeder, Bill Lorensen
+ *
+ *  For complete copyright, license and disclaimer of warranty information
+ *  please refer to the NOTICE file at the top of the ITK source tree.
+ *
+ *=========================================================================*/
+
+
+#ifndef itkTriangleMeshCurvatureCalculator_hxx
+#define itkTriangleMeshCurvatureCalculator_hxx
+
+#include <memory> // For unique_ptr
+#include "itkTriangleMeshCurvatureCalculator.h"
+#include "itkObjectFactory.h"
+#include "itkMath.h"
+#include "vnl/vnl_cross.h"
+#include "vnl/vnl_math.h"
+
+namespace itk
+{
+
+/*-------------------------------------------------------*/
+template <typename TInputMesh>
+void
+TriangleMeshCurvatureCalculator<TInputMesh>::PrintSelf(std::ostream & os, Indent indent) const
+{
+  this->Superclass::PrintSelf(os, indent);
+  os << indent << "CurvatureType: " << this->m_CurvatureType << "\n";
+}
+
+/** Compute Curvature */
+template <typename TInputMesh>
+void
+TriangleMeshCurvatureCalculator<TInputMesh>::Compute()
+{
+  MeshConstPointer inputMesh = this->m_TriangleMesh;
+  if (inputMesh == nullptr)
+  {
+    itkExceptionMacro("First set the Input Triangle Mesh to perform computation");
+  }
+
+  itkDebugMacro(<< "Executing Curvature");
+
+  if (this->GetCurvatureType() == Self::CurvatureGauss)
+  {
+    this->GetGaussCurvature(inputMesh);
+  }
+  else
+  {
+    itkExceptionMacro("Only Gauss Curvature type available");
+    return;
```

Comment:
I guess this `return` statement would be flagged as unreachable by the compiler if the relevant flag is activated. So please remove it.